Listener/RequestListener.php

```diff
@@ -78,7 +68,47 @@ public function onCoreRequest(GetResponseEvent $event)
 
             $this->interactor->setApplicationName($this->newRelic->getName(), $this->newRelic->getLicenseKey(), $this->newRelic->getXmit());
         }
+    }
+
+    /**
+     * Set the name of the transaction
+     *
+     * @param GetResponseEvent $event
+     */
+    public function setTransactionName(GetResponseEvent $event)
+    {
+        if (null === $request = $this->validateRequest($event)) {
+            return;
+        }
+
+        $transactionName = $this->transactionNamingStrategy->getTransactionName($request);
 
         $this->interactor->setTransactionName($transactionName);
     }
+
+    /**
+     * Make sure that it is not a subrequest and that we ignore paths that should be ignored
+     *
+     * @param GetResponseEvent $event
+     *
+     * @return \Symfony\Component\HttpFoundation\Request|null Return a Request or null iff it is not a master request
+     */
+    protected function validateRequest(GetResponseEvent $event)
```

Comment:
This should return true or false, and the argument must be the request object itself.